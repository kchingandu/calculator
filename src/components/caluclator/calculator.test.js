import React from 'react';
import { mount } from 'enzyme';
import Calculator from './Cacluclator';

describe('calculator', () => {
  let calculator;

  beforeEach(() => {
    calculator = mount(<Calculator/>);
  });

  it('should render the display input field and numeric and operator buttons', () => {
    expect(calculator.html()).toMatchSnapshot();
  });

  it('should update the input field from props', () => {
    calculator.setProps({displayText: '1.2x1.2'});

    expect(calculator.find('[id="display"]').props().value).toEqual('1.2x1.2');
  });

  it('should update the scroll position of the input field to always show right most text', () => {
    const displayInput = calculator.find('[id="display"]').getDOMNode();

    Object.defineProperty(displayInput, 'scrollWidth', {get: () => 316});

    calculator.setProps({displayText: '123'});

    expect(displayInput.scrollLeft).toEqual(316);
  });
})
;