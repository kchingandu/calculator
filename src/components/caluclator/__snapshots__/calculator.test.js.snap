// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculator should render the display input field and numeric and operator buttons 1`] = `"<div id=\\"calculator\\"><input dir=\\"ltr\\" id=\\"display\\" value=\\"0\\"><button class=\\"calculator-button\\">7</button><button class=\\"calculator-button\\">8</button><button class=\\"calculator-button\\">9</button><button class=\\"calculator-button\\">รท</button><button class=\\"calculator-button\\">4</button><button class=\\"calculator-button\\">5</button><button class=\\"calculator-button\\">6</button><button class=\\"calculator-button\\">x</button><button class=\\"calculator-button\\">1</button><button class=\\"calculator-button\\">2</button><button class=\\"calculator-button\\">3</button><button class=\\"calculator-button\\">-</button><button class=\\"calculator-button\\">0</button><button class=\\"calculator-button\\">.</button><button class=\\"calculator-button\\">=</button><button class=\\"calculator-button\\">+</button><button id=\\"clear\\" type=\\"button\\" class=\\"calculator-button\\">c</button></div>"`;
